src/model.rs

```diff
@@ -283,6 +283,44 @@ impl Model {
             })
             .collect()
     }
+    fn Reassign(&mut self, Coord: Coordinate, mut grammars: HashMap<Coordinate, Grammar>, i: i32) -> HashMap<Coordinate, Grammar> {
```

Comment:
could you make this name lowercase? also, i don't think you need to pass in both the model and the grammars
```suggestion
    fn reassign(&mut self, Coord: Coordinate, mut grammars: HashMap<Coordinate, Grammar>, i: i32) -> HashMap<Coordinate, Grammar> {
    let grammars = self.get_session().grammars;
```